pyserini/encode/__main__.py

```diff
@@ -22,19 +22,40 @@
 from pyserini.encode import UniCoilDocumentEncoder
 
 
-def init_encoder(encoder, device):
-    if 'dpr' in encoder.lower():
-        return DprDocumentEncoder(encoder, device=device)
-    elif 'tct_colbert' in encoder.lower():
-        return TctColBertDocumentEncoder(encoder, device=device)
-    elif 'ance' in encoder.lower():
-        return AnceDocumentEncoder(encoder, device=device)
-    elif 'sentence-transformers' in encoder.lower():
-        return AutoDocumentEncoder(encoder, device=device, pooling='mean', l2_norm=True)
-    elif 'unicoil' in encoder.lower():
-        return UniCoilDocumentEncoder(encoder, device=device)
+encoder_class_map = {
+    "dpr": DprDocumentEncoder,
+    "tct_colbert": TctColBertDocumentEncoder,
+    "ance": AnceDocumentEncoder,
+    "sentence-transformers": AutoDocumentEncoder,
+    "unicoil": UniCoilDocumentEncoder,
+    "auto": AutoDocumentEncoder,
+}
+
+def init_encoder(encoder, encoder_class, device):
+    _encoder_class = encoder_class
```

Comment:
if `encoder_class` is not provided, it will behave the same as previous `init_encoder`, otherwise it would just use the given `encoder_class`